single-product.html

```diff
@@ -0,0 +1,673 @@
+                    </div>
+                    <div class="mech-keyboard-img-container">
+                        <img class="mech-keyboard-img" src="../Images/multicolor-keyboard.jpg"
+                            alt="multicolor-mechanical-keyboard">
+                    </div>
+                    <div class="ecom-card-info">
+                        <h1 class="keyboard-name-heading">Fiodio Mechanical Keyboard</h1>
+                        <div class="rate-review">
+                            <div class="rate">4.3<strong class="star-symbol">✰</strong></div>
+                            <div class="review">1.2k reviews</div>
+                            <div class="icon-wishlist">
+                                <a class="icon-wishlist-link" href="#"><i class="fas fa-heart"></i></a>
+                            </div>
+                        </div>
+                        <div class="keyboard-prices">
+                            <div class="discount-price">
+                                ₹2499
+                            </div>
+                            <div class="total-price">
+                                <s>₹3999</s>
+                            </div>
+                        </div>
+                        <div class="buttons">
+                            <button class="btn btn-primary">
+                                <i class="bi bi-cart3"></i>
+                                <span class="icon-label-gap">Add to cart</span>
+                            </button>
+                        </div>
+                    </div>
+                </div>
+                <div class="e-commerce-card">
+                    <div class="badge-on-card">
+                        20% OFF
+                    </div>
+                    <div class="mech-keyboard-img-container">
+                        <img class="mech-keyboard-img" src="../Images/multicolor-keyboard.jpg"
+                            alt="multicolor-mechanical-keyboard">
+                    </div>
+                    <div class="ecom-card-info">
+                        <h1 class="keyboard-name-heading">Fiodio Mechanical Keyboard</h1>
+                        <div class="rate-review">
+                            <div class="rate">4.3<strong class="star-symbol">✰</strong></div>
+                            <div class="review">1.2k reviews</div>
+                            <div class="icon-wishlist">
+                                <a class="icon-wishlist-link" href="#"><i class="fas fa-heart"></i></a>
+                            </div>
+                        </div>
+                        <div class="keyboard-prices">
+                            <div class="discount-price">
+                                ₹2499
+                            </div>
+                            <div class="total-price">
+                                <s>₹3999</s>
+                            </div>
+                        </div>
+                        <div class="buttons">
```

Comment:
Changes: The class "buttons" is not doing anything to your button also don't use the button inside of div if not necessary, breaks the semantic HTML.

Reply:
> Changes: The class "buttons" is not doing anything to your button also don't use the button inside of div if not necessary, breaks the semantic HTML.

@zoef-yt any reasons not to use buttons inside div. any references do you have? even I have used the same structure, 😁

Reply:
Yep, same question, I need to read about this.